tools/benchmark/src/test/CustomOptionBenchmark.test.ts

```diff
@@ -17,3 +17,30 @@ describe("`benchmarkCustom` function", () => {
 		});
 	});
 });
+
+describe("BenchmarkCustom error handling", () => {
+	const expectedErrorMessage = "INTENTIONAL error to test error handling";
+	let benchmarkEndPayloadIsCorrect: boolean = false;
+
+	const testObject = benchmarkCustom({
+		title: `test`,
+		type: BenchmarkType.Measurement,
+		run: async () => {
+			throw new Error(expectedErrorMessage);
+		},
+	});
+
+	testObject.on("benchmark end", (benchmarkError: BenchmarkError) => {
+		if (benchmarkError.error === expectedErrorMessage) {
+			benchmarkEndPayloadIsCorrect = true;
+		}
+	});
+
+	afterEach(function () {
+		if (benchmarkEndPayloadIsCorrect) {
+			this.skip();
+		} else {
+			throw new Error("The benchmark end payload was not correct.");
+		}
```

Comment:
Let's add a good comment here about why we're doing this very unusual thing, and clarifying that this test getting skipped is actually the expected behavior when everything went right.